neuro_extras/image.py

```diff
@@ -240,42 +319,63 @@ async def _build_image(
     async with get_neuro_client(cluster=cluster) as client:
         context_uri = client.parse.str_to_uri(
             context,
-            allowed_schemes=("file", "storage"),
+            allowed_schemes=("file",) if local else ("file", "storage"),
         )
         image_exists = await _check_image_exists(image_uri_str, client)
-        if image_exists and force_overwrite:
-            logger.warning(
-                f"Target image '{image_uri_str}' exists and will be overwritten."
-            )
-        elif image_exists and not force_overwrite:
-            raise click.ClickException(
-                f"Target image '{image_uri_str}' exists. "
-                f"Use -F/--force-overwrite flag to enforce overwriting."
-            )
+        if image_exists:
+            if force_overwrite:
+                logger.warning(
+                    f"Target image '{image_uri_str}' exists and will be overwritten."
+                )
+            else:
+                raise click.ClickException(
+                    f"Target image '{image_uri_str}' exists. "
+                    f"Use -F/--force-overwrite flag to enforce overwriting."
+                )
 
         builder = ImageBuilder(
             client, extra_registry_auths=registry_auths, verbose=verbose
         )
-        exit_code = await builder.build(
-            dockerfile_path=dockerfile_path,
-            context_uri=context_uri,
-            image_uri_str=image_uri_str,
-            use_cache=use_cache,
-            build_args=build_args,
-            volumes=volume,
-            envs=env,
-            job_preset=preset,
-            build_tags=build_tags,
-        )
+        if local:
+            exit_code = await builder.build_local(
+                dockerfile_path=dockerfile_path,
+                context_uri=context_uri,
+                image_uri_str=image_uri_str,
+                build_args=build_args,
+                build_tags=build_tags,
+            )
+        else:
+            exit_code = await builder.build(
+                dockerfile_path=dockerfile_path,
+                context_uri=context_uri,
+                image_uri_str=image_uri_str,
+                use_cache=use_cache,
+                build_args=build_args,
+                volumes=volume,
+                envs=env,
+                job_preset=preset,
+                build_tags=build_tags,
+            )
         if exit_code == EX_OK:
             logger.info(f"Successfully built {image_uri_str}")
+            if local:
+                logger.info(f"Pushing image to registry")
+                remote_image = await _parse_neuro_image(image_uri_str)
+                console = Console()
+                progress = DockerImageProgress.create(
+                    console=console, quiet=not verbose
+                )
+                local_image = client.parse.local_image(remote_image.as_docker_url())
+                await client.images.push(local_image, remote_image, progress=progress)
```

Comment:
If it fails, we might suggest user to retry the image push via `neuro image push {local-name} {remote-name}`. Or retry is embedded into this client method? (afaik, it is present only in CLI).